src/kbmod_wf/single_chip_workflow2.py

```diff
@@ -0,0 +1,197 @@
+    ignore_for_cache=["logging_file"],
+)
+def step2(inputs=(), outputs=(), runtime_config={}, logging_file=None):
+    from kbmod_wf.utilities.logger_utilities import get_configured_logger, ErrorLogger
+    logger = get_configured_logger("kbmod.search_task", logging_file.filepath)
+
+    import json
+    
+    from kbmod.work_unit import WorkUnit
+    from kbmod.run_search import SearchRunner
+
+    with ErrorLogger(logger):
+        wu = WorkUnit.from_fits(inputs[0].filename)
+        res = SearchRunner().run_search_from_work_unit(wu)
+        header = wu.wcs.to_header(relax=True)
+        h, w = wu.wcs.pixel_shape
+        header["NAXIS1"], header["NAXIS2"] = h, w
+        res.table.meta["wcs"] = json.dumps(dict(header))
+        res.write_table(outputs[0].filename)
+
+    return outputs
+
+
+def workflow_runner(env=None, runtime_config={}):
+    """This function will load and configure Parsl, and run the workflow.
+
+    Parameters
+    ----------
+    env : str, optional
+        Environment string used to define which resource configuration to use,
+        by default None
+    runtime_config : dict, optional
+        Dictionary of assorted runtime configuration parameters, by default {}
+    """
+    resource_config = get_resource_config(env=env)
+                              
+    resource_config = apply_runtime_updates(resource_config, runtime_config)
+                              
+
+    app_configs = runtime_config.get("apps", {})
+                              
+
+    dfk = parsl.load(resource_config)
+                              
+    if dfk:
+        logging_file = File(os.path.join(dfk.run_dir, "kbmod.log"))
+        logger = get_configured_logger("workflow.workflow_runner", logging_file.filepath)
+
+        if runtime_config is not None:
+            logger.info(f"Using runtime configuration definition:\n{toml.dumps(runtime_config)}")
+
+        logger.info("Starting workflow")
+        
+        directory_path = runtime_config.get("staging_directory", ".")
+        file_pattern = runtime_config.get("file_pattern", "*.collection")
+        pattern = os.path.join(directory_path, file_pattern)
+        entries = glob.glob(pattern)
+        logger.info(f"Found {len(entries)} files in {directory_path}")
+
+        step1_futures = []
+        for collection in entries:
+            collection_file = File(collection)
+            collname = os.path.basename(collection)
+            collname = collname.split(".")[0]
+            step1_futures.append(
+                step1(
+                    inputs=[collection_file],
+                    outputs=[File(f"{collname}_resampled.wu")],
+                    runtime_config=app_configs.get("ic_to_wu", {}),
+                    logging_file=logging_file,
+                )
+            )
+
+        # run kbmod search on each reprojected WorkUnit
+        search_futures = []
+        for resampled_future in step1_futures:
+            search_futures.append(
+                step2(
```

Comment:
You should be able to use the workflow_tasks/kbmod_search defined here: https://github.com/dirac-institute/kbmod-wf/blob/main/src/kbmod_wf/workflow_tasks/kbmod_search.py I think, yeah?

Reply:
If in the previous step I create a sharded WU, write it to disk, reproject it and then write the shards down to disk, then kbmod_search would work out of the box. Since I don't do that (having already loaded all the images to memory when creating the first WU from IC) I just write one reprojected non-sharded WU. It will have to be completely loaded to perform the search again anyhow, so the search step doesn't benefit from sharding.